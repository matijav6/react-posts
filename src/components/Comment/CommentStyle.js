import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  min-width: 0;
  max-width: 450px;
  word-wrap: break-word;
  background-color: #fff;
  background-clip: border-box;
  border: 1px solid rgba(0, 0, 0, 0.125);
  border-radius: 0.25rem;
  box-shadow: 0 0.125rem 0.25rem rgba(0, 0, 0, 0.075);
  overflow: hidden;
`;

export const Body = styled.div`
  display: flex;
  flex-direction: column;
  flex: 1 1 auto;
  min-height: 1px;
  padding: 1.25rem;
`;

export const Title = styled.h5`
  margin: 15px 0;
  max-width: 65%;
`;

export const Text = styled.p`
  font-size: 13px;
  color: #6c757d;
`;

export const Author = styled.p`
  line-height: 25px;
  color: #6c757d;
`;
