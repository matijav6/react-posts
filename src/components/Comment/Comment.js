import React from 'react';
import classNames from 'classnames';
import { Container, Body, Title, Text, Author } from './CommentStyle';

export const Comment = ({classes, children, ...restProps}) => {
    return (
        <Container className={classNames('comment', classes)} {...restProps}>
            {children}
        </Container>
    );
};


Comment.Body = function CommentBody({ classes, children, ...restProps }) {
    return (
        <Body className={classNames('comment__body', classes)} {...restProps}>
            {children}
        </Body>
    );
};

Comment.Title = function CommentTitle({ classes, children, ...restProps }) {
    return (
        <Title className={classNames('comment__title', classes)} {...restProps}>
            {children}
        </Title>
    );
};

Comment.Text = function CommentText({ classes, children, ...restProps }) {
    return (
        <Text className={classNames('comment__text', classes)} {...restProps}>
            {children}
        </Text>
    );
};

Comment.Author = function CommentAuthor({ classes, children, ...restProps }) {
    return (
        <Author className={classNames('comment__author', classes)} {...restProps}>
            {children}
        </Author>
    );
};
