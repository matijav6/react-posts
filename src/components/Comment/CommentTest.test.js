import renderer from 'react-test-renderer';
import {Comment} from './Comment'

test('Comment renders correctly', () => {
    const tree = renderer
        .create(<Comment/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
