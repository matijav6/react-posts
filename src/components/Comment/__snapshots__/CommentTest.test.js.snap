// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment renders correctly 1`] = `
<div
  className="sc-bdfBwQ ilTjGw comment"
/>
`;
